packages/react-core/src/components/DualListSelector/DualListSelector.tsx

```diff
@@ -217,13 +322,131 @@ export class DualListSelector extends React.Component<DualListSelectorProps, Dua
         availableOptionsSelected: isChosen ? prevState.availableOptionsSelected : updatedArray
       };
     });
+    const onOptionSelect = this.props.onOptionSelect;
+    if (onOptionSelect) {
+      onOptionSelect(e);
+    }
+  };
+
+  onTreeOptionSelect = (
+    e: React.MouseEvent | React.ChangeEvent,
+    index: number,
+    isChosen: boolean,
+    text?: string,
+    itemData?: any,
+    parentData?: any
+  ) => {
+    this.setState(prevState => {
+      const selectedOptions = isChosen ? prevState.chosenTreeOptionsSelected : prevState.availableTreeOptionsSelected;
+
+      let updatedArray = null;
+      if (itemData.children) {
+        const panelOptions = isChosen ? this.state.chosenOptions : this.state.availableOptions;
+        const selectedOptionTree = panelOptions
+          .map(opt => Object.assign({}, opt))
+          .filter(item => filterTreeItems(item as DualListSelectorTreeItemData, [text]));
+        const flatSelectedItems = flattenTreeWithFolders(selectedOptionTree as DualListSelectorTreeItemData[]);
 
+        if (selectedOptions.includes(text)) {
```

Comment:
seems this might be a bit brittle? considering you could have multiple branches with the same named nodes? clicking one would auto-check the other
<img width="375" alt="Screen Shot 2020-12-08 at 10 06 39 AM" src="https://user-images.githubusercontent.com/869106/101501055-4185ad00-393d-11eb-8a90-6b9f9eaecf6a.png">

Reply:
Another example: In this case, selecting `Option Z` highlights both parent folders. Maybe this is an edge case but perhaps we shouldn't key off of text for identifying a node
<img width="369" alt="Screen Shot 2020-12-08 at 10 12 36 AM" src="https://user-images.githubusercontent.com/869106/101501875-439c3b80-393e-11eb-9fb8-1c2bc7c02b1a.png">

Reply:
Good point. Maybe I should key off of `id` or a similar prop instead.